source/common/http/http2/metadata_decoder.cc

```diff
@@ -1,12 +1,46 @@
 #include "source/common/http/http2/metadata_decoder.h"
 
 #include "source/common/common/assert.h"
+#include "source/common/runtime/runtime_features.h"
 
 #include "absl/container/fixed_array.h"
+#include "quiche/http2/decoder/decode_buffer.h"
+#include "quiche/http2/hpack/decoder/hpack_decoder.h"
+#include "quiche/http2/hpack/decoder/hpack_decoder_listener.h"
 
 namespace Envoy {
 namespace Http {
 namespace Http2 {
+namespace {
+
+class QuicheDecoderListener : public http2::HpackDecoderListener {
+public:
+  explicit QuicheDecoderListener(MetadataMap& map) : map_(map) {}
+  void OnHeaderListStart() override {}
+
+  void OnHeader(const std::string& name, const std::string& value) override {
+    map_.emplace(name, value);
```

Comment:
Consider using `push_back()` instead of `emplace()` per https://abseil.io/tips/112.

Reply:
I would slightly prefer to leave `emplace()`, for parity with the similar call within `decodeMetadataPayloadUsingNghttp2()`.